engines/kyra/engine/eobcommon.cpp

```diff
@@ -364,14 +378,26 @@ Common::KeymapArray EoBCoreEngine::initKeymaps(const Common::String &gameId) {
 	};
 
 	for (uint i = 0; i < ARRAYSIZE(keyActionEntries); ++i) {
-		Common::Action *const act = new Common::Action(keyActionEntries[i].id, keyActionEntries[i].description);
+		act = new Common::Action(keyActionEntries[i].id, keyActionEntries[i].description);
 		act->setKeyEvent(keyActionEntries[i].ks);
 		act->addDefaultInputMapping(keyActionEntries[i].defaultHwId);
+		if (!strcmp(keyActionEntries[i].defaultHwId, "UP"))
+			act->addDefaultInputMapping("JOY_UP");
+		else if (!strcmp(keyActionEntries[i].defaultHwId, "DOWN"))
+			act->addDefaultInputMapping("JOY_DOWN");
+		else if (!strcmp(keyActionEntries[i].defaultHwId, "LEFT"))
+			act->addDefaultInputMapping("JOY_LEFT_TRIGGER");
+		else if (!strcmp(keyActionEntries[i].defaultHwId, "RIGHT"))
+			act->addDefaultInputMapping("JOY_RIGHT_TRIGGER");
+		else if (!strcmp(keyActionEntries[i].defaultHwId, "HOME"))
+			act->addDefaultInputMapping("JOY_LEFT");
+		else if (!strcmp(keyActionEntries[i].defaultHwId, "PAGEUP"))
+			act->addDefaultInputMapping("JOY_RIGHT");
```

Comment:
This is not very good style. I suggest you change `Common::KeyActionEntry` to add a new field `defaultJoyHwId` (it's only used in the kyra engine). Alternatively you can expand the definition of the actions not to use `KeyActionEntry` like in the other keymaps. The struct needlessly constrains the actions' definitions.

Reply:
Ok I expanded the definitions.